network/src/main/scala/com/linkedin/norbert/network/netty/DarkCanaryChannelHandler.scala

```diff
@@ -171,6 +171,7 @@ class DarkCanaryChannelHandler extends Logging {
                       request.outputSerializer,
                       None,
                       0)
+                    request.headers.foreach{kv_pair => newRequest.addHeader(kv_pair._1, kv_pair._2)}
```

Comment:
I think that we had better create some utility function called "CreateDarkRequest" which includes the request substantiation and settings in all together.